### PR TITLE
The default branch "master" is now configurable

### PR DESCRIPTION
**Pull request by Adysone**
Creates a new configuration variable 'default_branch' to fill automatically the "branch" input when creating a repository on MantisBT.

### CODE REVIEW

File: SourceGitlab/SourceGitlab.php [hunk 1]
```diff
@@ -115,7 +115,7 @@ public function update_repo_form( $p_repo ) {
 		if ( isset( $p_repo->info['master_branch'] ) ) {
 			$t_master_branch = $p_repo->info['master_branch'];
 		} else {
-			$t_master_branch = 'master';
+			$t_master_branch = config_get('plugin_Source_default_branch', 'master');
```
Comment (dregad) [inline SourceGitlab/SourceGitlab.php]: coding guidelines

File: SourceGitlab/SourceGitlab.php [hunk 2]
```diff
@@ -249,7 +249,7 @@ public function import_full( $p_repo ) {
 
 		$t_branch = $p_repo->info['master_branch'];
 		if ( is_blank( $t_branch ) ) {
-			$t_branch = 'master';
+			$t_branch = config_get('plugin_Source_default_branch', 'master');
```
Comment (dregad) [inline SourceGitlab/SourceGitlab.php]: coding guidelines

**Comment by Adysone** [comment]
All should be good now. I can't test other VCS than GitLab but it should be fine. Thanks for your answer.

**Comment by Adysone** [comment]
> The new _default_branch_ setting only applies to Git-based repositories, and does not make any sense in the context of other repo types (e.g. SVN). However, since the setting is global, the config page always displays it, which could be confusing to users of other VCS especially if they don't use any Git-based repositories.
> 
> IMO, the new setting should clearly indicate that it only applies to Git-based plugin types, and ideally should only be available if one of these is installed.
> 
> Moreover, it would be good to uniformize the labelling (the setting is called _Primary Branch_ in the child plugins), and give an indication that it can be a comma-separated list of branches (or `*`).

I can't check if a plugin is installed (I don't know Mantis API enough), but I improved labelling. 👍 

> I would also advise to avoid code duplication, and retrieve the default value via the _MantisSourceGitBasePlugin_ abstract class. Maybe the display on config page could be handled from there as well, although that might be a bit tricky (might need creation of a new internal plugin event).

I can't do that, I'm not good enough. 😄 

If all I did is not enough, nevermind, you can close que PR, I will use my fork as plugin. At least I tried 😉 

Thank you for your interest btw.

**Comment by dregad** [comment]
@Adysone I implemented your feature into the Git base plugin class as per my earlier comment. This means that your original code is all but gone, but thanks anyway for your contribution.

**Pull request closed without merging**